doc/source/roadmap-detailed.rst

```diff
@@ -0,0 +1,385 @@
+    with Cython memoryviews.
+
+Regarding build environments:
+
+  - SciPy builds from source on Windows now with a MSVC + MinGW-w64 gfortran
+    toolchain, which we're using for official releases.
+    MSVC + Intel Fortran + MKL works as well, and is easier for users (as long
+    as they have access to ifort and MKL of course).  This mainly needs better
+    documentation at the moment.
+  - We're aiming to gradually increase the minimum version of LAPACK that is
+    required, so we can use newer features.  Support for Accelerate on macOS
+    has been dropped.  We do rely quite heavily on OpenBLAS, and its stability
+    is a worry (often only one of the recent releases works without test
+    failures) - improvements in testing and build documentation at least are
+    needed.
+
+Continuous integration is in good shape, it covers Windows, macOS and Linux, as well
+as a range of versions of our dependencies and building release quality wheels.
+
+
+Modules
+-------
+
+cluster
+```````
+This module is in good shape.
+
+
+constants
+`````````
+This module is basically done, low-maintenance and without open issues.
+
+
+fftpack
+```````
+We aim to follow NumPy in adopting ``pocketfft`` (see `this NumPy PR
+<https://github.com/numpy/numpy/pull/11888>`__).  That will address a number of
+maintenance issues, and increase performance (both accuracy and speed).
+Of particular interest regarding performance is the Bluestein algorithm (or
+chirp Z-transform), which we have been wanting to add to ``fftpack`` for a long
+time.
+
+We probably want to deprecate fftpack.convolve as public function (it was not
+meant to be public).
+
+There's a large overlap with ``numpy.fft``.  This duplication has to change
+(both are too widely used to deprecate one); in the documentation we should
+make clear that ``scipy.fftpack`` is preferred over ``numpy.fft``.
+If there are differences in signature or functionality, the best version
+should be picked case by case (example: numpy's ``rfft`` is preferred, see
+`gh-2487 <https://github.com/scipy/scipy/issues/2487>`__).
+
+Ideas for new features:
+
+- Add a backend/plugin system.  At the moment pyFFTW is monkeypatching SciPy,
+  and ``mkl_fft`` provides ``fftpack``-compatible functions as well.  We should
+  provide a method to support such packages.
+
+integrate
+`````````
+Needed for ODE solvers:
+
+  - Documentation is pretty bad, needs fixing
+  - A new ODE solver interface  (``solve_ivp``) was added in SciPy 1.0.0.
+    In the future we can consider (soft-)deprecating the older API.
+
+The numerical integration functions are in good shape.  Support for integrating
+complex-valued functions and integrating multiple intervals (see `gh-3325
+<https://github.com/scipy/scipy/issues/3325>`__) could be added.
+
+
+interpolate
+```````````
+
+Ideas for new features:
+
+  - Spline fitting routines with better user control.
+  - Integration and differentiation and arithmetic routines for splines
```

Comment:
Integration and differentiation are implemented, not sure what is meant by arithmetic routines.

Reply:
agree, removed